rpc/src/v1/helpers/dispatch.rs

```diff
@@ -449,28 +477,61 @@ fn sign_transaction(
 #[derive(Debug, Clone, Copy)]
 enum ProspectiveSignerState {
 	TryProspectiveSign,
+	PostSign,
+	WaitForPostSign,
 	WaitForNonce,
-	Finish,
 }
 
-struct ProspectiveSigner {
+struct ProspectiveSigner<P: PostSign> {
 	accounts: Arc<AccountProvider>,
 	filled: FilledTransactionRequest,
 	chain_id: Option<u64>,
 	reserved: nonce::Reserved,
 	password: SignWith,
 	state: ProspectiveSignerState,
-	prospective: Option<Result<WithToken<SignedTransaction>>>,
+	prospective: Option<WithToken<SignedTransaction>>,
 	ready: Option<nonce::Ready>,
+	post_sign: Option<P>,
```

Comment:
I know that it's aligned with how currently the state is passed in the `ProspectiveSigner`, but we might consider refactoring this in a way that the required items are passed in the `ProspectiveSignerState` enum instead to avoid so many `Option<` and `.expects()`.
Not required for this PR (or not required at all), unless someone wants to pick it up :)